bluesky/run_engine.py

```diff
@@ -1343,13 +1343,16 @@ def _collect(self, msg):
         self._uncollected.remove(obj)
         stream_name = self._flyer_stream_names.pop(obj)
 
-        # TODO Since Flyer.describe() return a *list* of data_keys it should
-        # probably have a different method name (describe_all?) *and* have some
-        # way of injecting a 'name' like the create msg does. For now, the name
-        # is hard-coded to None.
-        data_keys_list = obj.describe()
+        description = obj.describe_collect()  # a dict or a list
+        if hasattr(description, 'items'):
```

Comment:
Below, we actually *call* `description.items()` so there is zero harm in performing this check.